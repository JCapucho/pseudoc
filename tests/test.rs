use std::fs::read_to_string;

#[test]
fn insta_test() {
    insta::glob!("*.psc", |file| {
        let code = read_to_string(file).unwrap();

        let mut rodeo = pseudoc::new_rodeo();
        let parse_result = pseudoc::build_parse(&code, &mut rodeo);
        let resolver = rodeo.into_resolver();

        let default_name = "test";

        match parse_result {
            Ok(ref parse) => {
                let mut buf = Vec::new();

                pseudoc::build_pseudo(parse, &resolver, default_name, &mut buf).unwrap();
                buf.push(b'\n');
                pseudoc::build_pascal(parse, &resolver, default_name, &mut buf).unwrap();
                buf.push(b'\n');
                pseudoc::build_dot(parse, &resolver, &mut buf).unwrap();
                buf.push(b'\n');
                let out = std::str::from_utf8(buf.as_slice()).unwrap().to_string();
                insta::assert_snapshot!(out);
            },
            Err(errors) => {
                insta::assert_debug_snapshot!(errors);
            },
        }
    });
}
