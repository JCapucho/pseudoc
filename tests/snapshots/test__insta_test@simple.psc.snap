---
source: tests/test.rs
expression: out
input_file: tests/simple.psc

---
Algoritmo soma;
VariÃ¡vel
    a, b, soma: inteiro;
InÃ­cio
    a ğŸ ” 3;
    b ğŸ ” 2;
    soma ğŸ ” a + b;
Fim.

Program soma;
Var
    a, b, soma: integer;
Begin
    a := 3;
    b := 2;
    soma := a + b;
End.

digraph "soma" {
start [shape=oval, label="InÃ­cio"]
end [shape=oval, label="Fim"]
stmt_0_0 [label = "a = 3", shape=box]
stmt_0_0 -> stmt_0_1
stmt_0_1 [label = "b = 2", shape=box]
stmt_0_1 -> stmt_0_2
stmt_0_2 [label = "soma = a + b", shape=box]
stmt_0_2 -> end

start -> stmt_0_0
}


